examples/catalyst_contrastive/barlow_twins.py

```diff
@@ -0,0 +1,131 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from torch.utils.data import DataLoader
+import torchvision
+from torchvision import transforms
+from torchvision.models.resnet import resnet50
+
+from catalyst import dl
+from catalyst.contrib.nn import BarlowTwinsLoss
+
+
+class CifarPairTransform:
+    def __init__(self, train_transform=True, pair_transform=True):
+        if train_transform is True:
+            self.transform = transforms.Compose(
+                [
+                    transforms.RandomResizedCrop(32),
+                    transforms.RandomHorizontalFlip(p=0.5),
+                    transforms.RandomApply([transforms.ColorJitter(0.4, 0.4, 0.4, 0.1)], p=0.8),
+                    transforms.RandomGrayscale(p=0.2),
+                    transforms.ToTensor(),
+                    transforms.Normalize([0.4914, 0.4822, 0.4465], [0.2023, 0.1994, 0.2010]),
+                ]
+            )
+        else:
+            self.transform = transforms.Compose(
+                [
+                    transforms.ToTensor(),
+                    transforms.Normalize([0.4914, 0.4822, 0.4465], [0.2023, 0.1994, 0.2010]),
+                ]
+            )
+        self.pair_transform = pair_transform
+
+    def __call__(self, x):
+        if self.pair_transform is True:
+            y1 = self.transform(x)
+            y2 = self.transform(x)
+            return y1, y2
+        else:
+            return self.transform(x)
+
+
+class Model(nn.Module):
+    def __init__(self, feature_dim=128):
+        super(Model, self).__init__()
+
+        self.f = []
+        for _, module in resnet50().named_children():
+            if not isinstance(module, nn.Linear) and not isinstance(module, nn.MaxPool2d):
+                self.f.append(module)
+
+        # encoder
+        self.f = nn.Sequential(*self.f)
```

Comment:
not sure if, but is it possible to use something like
- https://github.com/catalyst-team/catalyst/blob/master/catalyst/contrib/models/cv/encoders/resnet.py
- or https://github.com/catalyst-team/introspection/blob/main/introspection/modules.py#L49#L52

to simplify the filtering?

Reply:
Thanks, I will check this.

Reply:
[Done](https://github.com/catalyst-team/catalyst/pull/1261/commits/f1d6e50d32a09da6f8892244778dee946950e432)